### PR TITLE
BAEL-4431-jdbcmetadata: Article about extracting metadata using JDBC

### PR DESCRIPTION
**Pull request by sahilsingla112**
http://jira.baeldung.com/browse/BAEL-4431
http://inprogress.baeldung.com/?p=222927&preview=true
http://inprogress.baeldung.com/wp-admin/post.php?post=222927&action=edit&classic-editor&classic-editor__forget

### CODE REVIEW

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/JdbcMetadataApplication.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            metadataExtractor.extractSystemTables();
+            metadataExtractor.extractViews();
+            String tableName = "CUSTOMER";
+            metadataExtractor.extractColumnInfo(tableName);
+            metadataExtractor.extractPrimaryKeys(tableName);
+            metadataExtractor.extractForeignKeys("CUST_ADDRESS");
+            metadataExtractor.extractDatabaseInfo();
+            metadataExtractor.extractUserName();
+            metadataExtractor.extractSupportedFeatures();
+        }catch (SQLException e){
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/JdbcMetadataApplication.java:26]:
```suggestion
        } catch (SQLException e){
```

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class MetadataExtractor {
+	private final DatabaseMetaData databaseMetaData;
+
+	public MetadataExtractor(Connection connection) throws SQLException {
+		this.databaseMetaData = connection.getMetaData();
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:12]: This line is using tabs. Be sure to use the standard code formatter (available in the tutorials repo, under eclipse or intelliJ foder, depending on which IDE you're using)

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        DatabaseMetaData databaseMetaData = connection.getMetaData();
+	}
+	
+	public void extractTableInfo() throws SQLException {
+        ResultSet resultSet = databaseMetaData.getTables(null, null, "CUST%", new String[]{"TABLE"});
+        while(resultSet.next())
+        {
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:19]: Opening braces should be at the end of a line of code, not at the beginning of a line by itself. Several of the while statements are done like this.

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            String datatype = columns.getString("DATA_TYPE");
+            String isNullable = columns.getString("IS_NULLABLE");
+            String isAutoIncrement = columns.getString("IS_AUTOINCREMENT");
+            System.out.println(String.format("ColumnName: %s, columnSize: %s, datatype: %s, isColumnNullable: %s, isAutoIncrementEnabled: %s", columnName, columnSize, datatype, isNullable, isAutoIncrement));
+        }
+    }
+
+    public void extractPrimaryKeys(String tableName) throws SQLException{
+	    ResultSet primaryKeys = databaseMetaData.getPrimaryKeys(null, null, tableName);
+        while(primaryKeys.next()){
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:60]:
```suggestion
        while(primaryKeys.next()) {
```

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    public void fun() throws SQLException{
+
+    }
+
+
+    public void extractForeignKeys(String tableName) throws SQLException{
+        ResultSet foreignKeys = databaseMetaData.getImportedKeys(null, null, tableName);
+        while(foreignKeys.next()){
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:74]:
```suggestion
        while(foreignKeys.next()) {
```

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            String pkTableName = foreignKeys.getString("PKTABLE_NAME");
+            String fkTableName = foreignKeys.getString("FKTABLE_NAME");
+            String pkColumnName = foreignKeys.getString("PKCOLUMN_NAME");
+            String fkColumnName = foreignKeys.getString("FKCOLUMN_NAME");
+            System.out.println(String.format("pkTableName:%s, fkTableName:%s, pkColumnName:%s, fkColumnName:%s", pkTableName, fkTableName, pkColumnName, fkColumnName));
+        }
+    }
+
+    public void extractDatabaseInfo() throws SQLException{
+        String productName    = databaseMetaData.getDatabaseProductName();
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:84]: avoid the extra spaces

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+        System.out.println(String.format("Product name:%s, Product version:%s", productName, productVersion));
+        System.out.println(String.format("Driver name:%s, Driver Version:%s", driverName, driverVersion));
+    }
+
+    public void extractUserName() throws SQLException{
+        String userName = databaseMetaData.getUserName();
+        System.out.println(userName);
+        ResultSet schemas = databaseMetaData.getSchemas();
+        while (schemas.next()){
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:98]:
```suggestion
        while (schemas.next()) {
```

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+            String table_schem = schemas.getString("TABLE_SCHEM");
+            String table_catalog = schemas.getString("TABLE_CATALOG");
+            System.out.println(String.format("Table_schema:%s, Table_catalog:%s", table_schem, table_catalog));
+        }
+    }
+
+    public void extractSupportedFeatures() throws SQLException{
```
Comment (KevinGilmore) [inline persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/MetadataExtractor.java:105]:
```suggestion
    public void extractSupportedFeatures() throws SQLException {
```

File: persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/JdbcMetadataApplication.java (hunk 1 above)
Comment (sahilsingla112) [reply persistence-modules/core-java-persistence/src/main/java/com/baeldung/jdbcmetadata/JdbcMetadataApplication.java]: @KevinGilmore  Handled all the review comments by using the recommended formatter.

**Pull request merged**